### PR TITLE
Fix releasing modified keys

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description
This PR fixes situation, when we release a key having modifier released after the same key has been pressed.

Consider the following sequence:
1. press Shift
2. press KeyA (emitted with key `A`)
3. release Shift
4. release KeyA (emitted with key `a`)

Now we treat 'A' and 'a' as different keys, and 'A' become stuck.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has been tested where possible.
- [x] All code has been profiled where possible.

### CODE REVIEW

File: src/rust/lib/frp/src/io/keyboard.rs [hunk 1]
```diff
@@ -175,14 +174,61 @@ impl Default for Key {
...
+/// Structure representing key with its code.
+///
+/// For difference between keys and codes see
+/// https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code and
+/// https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key
+#[allow(missing_docs)]
+#[derive(Clone,Debug,Default)]
+pub struct KeyWithCode {
+    pub key: Key,
+    pub code: String,
```
Comment (wdanilo) [inline src/rust/lib/frp/src/io/keyboard.rs:190]: alignment

File: src/rust/lib/frp/src/io/keyboard.rs [hunk 2]
```diff
@@ -208,39 +254,53 @@ impl KeyboardModel {
...
-    pub fn press(&self, key:&Key) {
-        self.set.borrow_mut().insert(key.clone());
+    pub fn press(&self, KeyWithCode{key,code}:&KeyWithCode) -> Key {
+        let pressed_key_opt = self.pressed_code_to_key.borrow_mut().get(code).cloned();
+        let key             = if let Some(pressed_key) = pressed_key_opt {
+            pressed_key
+        } else {
+            self.pressed_code_to_key.borrow_mut().insert(code.clone(),key.clone());
+            key.clone()
+        };
```
Comment (wdanilo) [inline src/rust/lib/frp/src/io/keyboard.rs:268]: use unwrap_or_else instead

File: src/rust/lib/frp/src/io/keyboard.rs [hunk 3]
```diff
@@ -298,16 +358,15 @@ impl Keyboard {
         let network = frp::Network::new();
         let model   = KeyboardModel::default();
         let source  = KeyboardSource::new(&network);
-        frp::extend! { network
-            eval source.down ((key) model.press(key));
-            eval source.up   ((key) model.release(key));
-            down             <- source.down.map(|t|t.clone());
-            is_meta_down     <- any(&source.down,&source.up).map(f_!(model.is_meta_down()));
-            meta_release     <= source.down.gate(&is_meta_down).map(
+        frp::extend! { TRACE_ALL network
```
Comment (wdanilo) [inline src/rust/lib/frp/src/io/keyboard.rs:361]: trace all

**Review by wdanilo — APPROVED** [review]

File: src/rust/lib/frp/src/io/keyboard.rs (hunk 2 above)
Comment (farmaazon) [reply src/rust/lib/frp/src/io/keyboard.rs:268]: clever!

**Pull request merged**